controllers/controllers.go

```diff
@@ -197,24 +197,21 @@ func windowsNodeVersionChangePredicate() predicate.Funcs {
 	return predicate.Funcs{
 		CreateFunc: func(e event.CreateEvent) bool {
 			// catch Machine-backed Windows node upgrades as they are re-created
-			return e.Object.GetLabels()[core.LabelOSStable] == "windows" &&
-				e.Object.GetAnnotations()[metadata.VersionAnnotation] == version.Get()
+			return isWindowsNode(e.Object) && e.Object.GetAnnotations()[metadata.VersionAnnotation] == version.Get()
```

Comment:
While it is fine to introduce `isWindowsNode()` in this method, updating all the places where it can be used should be a separate commit. I would have suggested a separate PR but I know @sebsoto requested the change be made here and in my opinion it is out of scope but can live with it if it is a separate commit.

Reply:
separate commit is my preference